core/Rename.cpp

```diff
@@ -96,7 +104,59 @@ namespace olympia
             ev_rename_insts_.schedule();
         }
     }
+    void Rename::getAckFromROB_(const InstPtr & inst_ptr)
+    {
+        sparta_assert(inst_ptr->getStatus() == Inst::Status::RETIRED,
+                        "Get ROB Ack, but the inst hasn't retired yet!");
+
+        core_types::RegFile rf;
+        auto dests = inst_ptr->getDestOpInfoList();
+        if(dests.size() > 0){
+            rf = determineRegisterFile(dests[0]);   
+        }
+        else{
+            // in the case of no dest register, it will be a NOP
+            rf = core_types::RegFile::RF_INTEGER;
+        }
+        
+        // freeing references to PRF
+        auto dest_register_bit_mask = inst_ptr->getDestRegisterBitMask(rf);
+        auto src_register_bit_mask = inst_ptr->getSrcRegisterBitMask(rf);
+        for(unsigned int i = 0; i < sparta::Scoreboard::MAX_REGISTERS; i++){
+            if(dest_register_bit_mask.test(i)){
```

Comment:
Consider stashing the destination register number in the instruction instead of searching for it in the mask.

Reply:
Addressed in [fbe7264](https://github.com/riscv-software-src/riscv-perf-model/pull/40/commits/fbe7264a97b4aadea92369bdf93a544c3077551e)

[Link](https://github.com/AaronGChan/riscv-perf-model/blob/fbe7264a97b4aadea92369bdf93a544c3077551e/core/Inst.hpp#L35) to RenameData inner class